packaging/version.py

```diff
@@ -198,7 +198,12 @@ class Version(_BaseVersion):
 
     def __init__(self, version):
         # Validate the version and parse it into pieces
-        match = self._regex.search(version)
+        try:
+            match = self._regex.search(version)
+        except TypeError:
+            raise InvalidVersion(
+                "Invalid type: '{0}' is not a string".format(type(version))
```

Comment:
We could also just make ``InvalidVersion`` descend directly from ``Exception`` instead.